### PR TITLE
Cranelift backend update to fork of clif version 0.43.1

### PR DESCRIPTION
**Pull request by bjfish**
The corresponding update to fork is here (which should merge first):
https://github.com/wasmerio/cranelift/pull/4

# Description

# Review

- [x] Add a short description of the the change to the CHANGELOG.md file

### CODE REVIEW

**Comment by bjfish** [comment]
bors try

**Comment by bjfish** [comment]
bors try

File: lib/clif-backend/Cargo.toml [hunk 1]
```diff
@@ -10,15 +10,15 @@ readme = "README.md"
...
-cranelift-wasm = { package = "wasmer-clif-fork-wasm", version = "0.33" }
-target-lexicon = "0.4"
-wasmparser = "0.35.1"
-byteorder = "1.3"
-nix = "0.15"
+cranelift-native = { git = "https://github.com/wasmerio/cranelift", branch = "feature/merge-cranelift-0-43-1" }
+cranelift-codegen = { git = "https://github.com/wasmerio/cranelift", branch = "feature/merge-cranelift-0-43-1" }
+cranelift-entity = { git = "https://github.com/wasmerio/cranelift", branch = "feature/merge-cranelift-0-43-1" }
+cranelift-frontend = { git = "https://github.com/wasmerio/cranelift", branch = "feature/merge-cranelift-0-43-1" }
+cranelift-wasm = { git = "https://github.com/wasmerio/cranelift", branch = "feature/merge-cranelift-0-43-1" }
```
Comment (bjfish) [inline lib/clif-backend/Cargo.toml]: TODO Update these before merging

Comment (syrusakbary) [reply lib/clif-backend/Cargo.toml]: Question: it seems that `cranelift-entity` and `cranelift-codegen` custom forks are no longer needed? @bjfish

Comment (bjfish) [reply lib/clif-backend/Cargo.toml]: Yes, also cust `cranelift-native` should not be needed. We will update these before merging.

**Comment by bjfish** [comment]
bors try

**Pull request merged**